Sources/SwiftInspectorVisitors/Tests/ProtocolVisitorSpec.swift

```diff
@@ -160,6 +160,103 @@ final class ProtocolVisitorSpec: QuickSpec {
               expect(protocolInfo?.genericRequirements.first?.relationship) == .conformsTo
             }
           }
+
+          context("with associatedtypes") {
+            var associatedTypeNameToInfoMap: [String: AssociatedtypeInfo]?
+            beforeEach {
+              let content = """
+                public protocol Collection {
```

Comment:
I straight stole Swift's `Collection` declaration and associatedtypes since they were immensely complicated.